lib/vmdb/settings/validator.rb

```diff
@@ -167,6 +167,42 @@ def smtp(data)
 
         return valid, errors
       end
+
+      def workers(data)
+        valid, errors = true, []
+        # worker_settings expects a hash-like structure with nested keys: :config and :workers
+        data = {:config => {:workers => data}}
+        MiqWorker.descendants.each do |worker_class|
+          result, new_errors = validate_worker_request_limit(worker_class, data)
+          if result == false
+            valid = false
+            errors += new_errors
+          end
+        end
+
+        return valid, errors
+      end
+
+      def validate_worker_request_limit(worker_class, data)
+        valid = true
+        errors = []
+        worker_settings = worker_class.worker_settings(data)
+        cpu_request     = worker_settings.fetch(:cpu_request_percent, 0)
+        cpu_limit       = worker_settings.fetch(:cpu_threshold_percent, Float::INFINITY)
+        memory_request  = worker_settings.fetch(:memory_request, 0.bytes)
+        memory_limit    = worker_settings.fetch(:memory_threshold, Float::INFINITY.megabytes)
```

Comment:
I don't think these defaults are correct.  The user could specify `cpu_threshold_percent`, but `cpu_request_percent` would be inherited and not passed to the data.

Reply:
`worker_class.worker_settings(data)` handles inheritance. It doesn't matter where the value came from, right?  We just default a value if not found for itself or inherited.  This avoids ugly nil checks.

Reply:
I have fixed up the tests to properly test inherited and overridden values and discarding of nils and usage of these sensible "valid" values just so the `<` `>` will be valid.  Note, the code that takes these values and determines if they should patch the deployment also [ignores any nil values](https://github.com/ManageIQ/manageiq/blob/65a5229e02f05ff0c8cb10b825dd9221771e0119/app/models/miq_server/worker_management/monitor/kubernetes.rb#L51-L61) so this code follows that expectation.